stl/inc/deque

```diff
@@ -772,35 +790,63 @@ public:
         emplace_front(_STD move(_Val));
     }
 
+#ifdef __cpp_lib_containers_ranges
+    template <_Container_compatible_range<_Ty> _Rng>
+    void prepend_range(_Rng&& _Range) {
+        _Orphan_all();
+
+        const auto _Oldsize = _Mysize();
+        _Restore_old_size_guard<_Pop_direction::_Front> _Guard{this, _Oldsize};
+        if constexpr (_RANGES bidirectional_range<_Rng>) {
+            auto _ULast        = _RANGES _Get_final_iterator_unwrapped(_Range);
+            const auto _UFirst = _RANGES _Ubegin(_Range);
+            while (_UFirst != _ULast) {
+                _Emplace_front_internal(*--_ULast); // prepend in order
+            }
+        } else {
+            auto _UFirst      = _RANGES _Ubegin(_Range);
+            const auto _ULast = _RANGES _Uend(_Range);
+            for (; _UFirst != _ULast; ++_UFirst) {
+                _Emplace_front_internal(*_UFirst); // prepend flipped
+            }
+
+            const auto _Num = static_cast<difference_type>(_Mysize() - _Oldsize);
+            _STD reverse(begin(), begin() + _Num); // flip new stuff in place

```

Comment:
There appears to be a wording defect here.

WG21-N4910 [sequence.reqmts] [/95](https://eel.is/c++draft/sequence.reqmts#95), [/97](https://eel.is/c++draft/sequence.reqmts#97) says:

> `a.prepend_range(rg)`
> *Preconditions:* `T` is *Cpp17EmplaceConstructible* into `X` from `*ranges::begin(rg)`.
> *Remarks:* Required for `deque`, `forward_list`, and `list`.

However, there should be additional requirements for `deque` when the range is not `bidirectional_range`, similar to how [/41](https://eel.is/c++draft/sequence.reqmts#41) says:

> Preconditions: [...] For `vector` and `deque`, `T` is also *Cpp17MoveInsertable* into `X`, *Cpp17MoveConstructible*, *Cpp17MoveAssignable*, and swappable (16.4.4.3). [...]

Should we file an LWG issue?

Reply:
I already filed an issue for this didn't I? (Narrator: "No, he had not.") Graargh.

Reply:
This is LWG-3742.